src/lib.rs

```diff
@@ -4,46 +4,98 @@
+///
+/// This list is created from
+/// <https://clang.llvm.org/docs/ClangFormatStyleOptions.html#basedonstyle>
 #[derive(Debug, PartialEq)]
 pub enum ClangFormatStyle {
+    /// A style complying with [Chromium’s style guide](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/styleguide/styleguide.md)
     Chromium,
+    /// Use the default clang-format style
     Default,
+    /// clang-format will try to find the .clang-format file located in the closest parent directory of the current directory.
     File,
+    /// A style complying with the [GNU coding standards](https://www.gnu.org/prep/standards/standards.html)
+    ///
+    /// Since clang-format 11
+    GNU,
+    /// A style complying with [Google’s C++ style guide](https://google.github.io/styleguide/cppguide.html)
     Google,
+    /// A style complying with the [LLVM coding standards](https://llvm.org/docs/CodingStandards.html)
     Llvm,
+    /// A style complying with [Microsoft’s style guide](https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference)
+    ///
+    /// Since clang-format 9
+    Microsoft,
+    /// A style complying with [Mozilla’s style guide](https://firefox-source-docs.mozilla.org/code-quality/coding-style/index.html)
     Mozilla,
+    /// A style complying with [WebKit’s style guide](https://www.webkit.org/coding/coding-style.html)
     WebKit,
+    /// Specify a custom input to the `--style` argument of clang-format
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use clang_format::{clang_format_with_style, ClangFormatStyle};
+    /// # fn main() {
+    /// # let input = r#"
+    /// #     struct Test {
+    /// #         bool field;
+    /// #     };
+    /// # "#;
+    /// let style = ClangFormatStyle::Custom("{ BasedOnStyle: Mozilla, IndentWidth: 8 }".to_string());
+    /// # let output = clang_format_with_style(input, &style);
+    /// # assert!(output.is_ok());
+    /// # assert_eq!(output.unwrap(), "\nstruct Test\n{\n        bool field;\n};\n");
+    /// # }
+    /// ```
+    Custom(String),
 }
 
 impl ClangFormatStyle {
     /// Converts the enum ClangFormatStyle to a string that clang-format expects
-    fn as_str(&self) -> &'static str {
+    fn as_str(&self) -> &str {
         match self {
             Self::Chromium => "Chromium",
             // Will use clang-format default options
             Self::Default => "{}",
             // Will look in parent directories for a .clang-format file
             Self::File => "file",
+            Self::GNU => "GNU",
             Self::Google => "Google",
             Self::Llvm => "LLVM",
+            Self::Microsoft => "Microsoft",
             Self::Mozilla => "Mozilla",
             Self::WebKit => "WebKit",
+            // Custom style arguments
+            Self::Custom(custom) => custom.as_str(),
         }
     }
 }
 
 /// Describes which error spawning clang-format failed with
 #[derive(Debug)]
 pub enum ClangFormatError {
```

Comment:
wonder if we should use something more modern here like `anyhow`. Or just use the built in `io::Result` and put `expect()` on the ones that are unlikely to fail ?

Reply:
At least implementing the `Error` trait would be a good thing to do.
Especially the cases something like `StdInWriteFailure` should also contain the source io::Error.

IIRC the right way to go would be to use `thiserror` and not anyhow.
anyhow should only be used by consuming applications, not libraries.

Reply:
We could then probably replace a lot of the `if let Ok(x)` with `map_err(ClangFormatError::from)?`.

This would probably be a breaking change then though :thinking: 
So probably better to do this in a follow-up PR.

Reply:
Right, lets do this in another PR